### PR TITLE
[Refactor] NLog logging to Microsoft.Extensions.Logging.Abstractions; fixes #219

### LINKED ISSUE #219: [Replace] NLog with the Microsoft Logging abstractions
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-SDK
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
replace the current Nog logging implemenattion using Microsoft logging abstractions so a library user can inject their own logging library (Nlog, Serilog, etc._

### Steps to Reproduce
NA

### PR DESCRIPTION
**Pull request by samatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-SDK-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

- [Remove] net461 TMF since it is out-of-support
- [Refactor] (remove) dependency on NLog and replace with Microsoft logging abstractions
- [Rename] .Dto namespace to ,DTO and fix in whole SDK
- [Add] INestedElementTreeGenerator to support dependency injection

### CODE REVIEW

File: CDP4Dal/DAL/Dal.cs [hunk 1]
```diff
@@ -40,31 +40,41 @@ namespace CDP4Dal.DAL
     using CDP4Dal.Operations;
     using CDP4Dal.Composition;
     using CDP4Dal.Exceptions;
-    
-    using NLog;
-    
-    using Iteration = CDP4Common.DTO.Iteration;
+
+	using Microsoft.Extensions.Logging;
```
Comment (lxatrhea) [inline CDP4Dal/DAL/Dal.cs:44]: Seems like something is wrong with spacing here.
Also in line 47.

File: CDP4Dal/Operations/ThingTransaction.cs [hunk 2]
```diff
@@ -38,18 +38,29 @@ namespace CDP4Dal.Operations
...
-    /// The Transaction class contains all requests for the creations, updates, deletions of things
-    /// </summary>
-    public class ThingTransaction : IThingTransaction
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Logging.Abstractions;
+
+	/// <summary>
+	/// The Transaction class contains all requests for the creations, updates, deletions of things
+	/// </summary>
+	public class ThingTransaction : IThingTransaction
```
Comment (lxatrhea) [inline CDP4Dal/Operations/ThingTransaction.cs:48]: Also in this class some strange things with spacing.

**Review by lxatrhea — CHANGES_REQUESTED** [review]
Somehow spacing/tabbing is inconsistent in the files, or at least it seems like that is the case.

File: CDP4Common/Helpers/NestedElementTreeGenerator.cs [hunk 3]
```diff
@@ -32,47 +32,59 @@ namespace CDP4Common.Helpers
...
+	    /// </summary>
+	    private readonly ILogger<NestedElementTreeGenerator> logger;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="NestedElementTreeGenerator"/> class.
+		/// </summary>
+		/// <param name="loggerFactory">
+		/// The (injected) <see cref="ILoggerFactory"/> used to setup logging
+		/// </param>
+		public NestedElementTreeGenerator(ILoggerFactory loggerFactory = null)
```
Comment (antoineatrhea) [inline CDP4Common/Helpers/NestedElementTreeGenerator.cs]: weird spacing on this file

File: CDP4Common/Poco/NestedParameter.cs [hunk 4]
```diff
@@ -24,9 +24,7 @@
 
 namespace CDP4Common.EngineeringModelData
 {
-    using System.Linq;
-
-    using CDP4Common.Helpers;
+	using CDP4Common.Helpers;
```
Comment (antoineatrhea) [inline CDP4Common/Poco/NestedParameter.cs]: wrong spacing on using

**Review by lxatrhea — APPROVED** [review]

File: CDP4Dal/DAL/Dal.cs [hunk 5]
```diff
@@ -41,30 +41,33 @@ namespace CDP4Dal.DAL
     using CDP4Dal.Composition;
     using CDP4Dal.Exceptions;
     
-    using NLog;
-    
-    using Iteration = CDP4Common.DTO.Iteration;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Logging.Abstractions;
+
+	using Iteration = CDP4Common.DTO.Iteration;
```
Comment (antoineatrhea) [inline CDP4Dal/DAL/Dal.cs:47]: Weird spacing in the file

File: CDP4Dal/Operations/ThingTransaction.cs [hunk 6]
```diff
@@ -40,17 +40,23 @@ namespace CDP4Dal.Operations
...
-        private readonly List<Thing> addedThing;
+	    /// <summary>
+	    /// The <see cref="ILogger"/> used to log
+	    /// </summary>
+	    private readonly ILogger<ThingTransaction> logger;
+
+		/// <summary>
+		/// Gets the added <see cref="Thing"/>s
+		/// </summary>
+		private readonly List<Thing> addedThing;
```
Comment (antoineatrhea) [inline CDP4Dal/Operations/ThingTransaction.cs]: Weird spacing in the file

File: CDP4Dal/Permission/PermissionService.cs [hunk 7]
```diff
@@ -34,26 +34,34 @@ namespace CDP4Dal.Permission
...
-        public PermissionService(ISession session)
+	/// <summary>
+	/// The Permission Service class for the CDP4 application
+	/// </summary>
+	public class PermissionService : IPermissionService
+    {
+		/// <summary>
+		/// The <see cref="ILogger"/> used to log
+		/// </summary>
+		private readonly ILogger<PermissionService> logger;
```
Comment (antoineatrhea) [inline CDP4Dal/Permission/PermissionService.cs:49]: Weird spacing in the file

File: CDP4Dal/Session.cs [hunk 8]
```diff
@@ -46,24 +46,25 @@ namespace CDP4Dal
...
-
+	    /// <summary>
+	    /// The <see cref="ILogger"/> used to log
+	    /// </summary>
+	    private readonly ILogger<Session> logger;
+
+		/// <summary>
+		/// Executes just before data from an <see cref="OperationContainer"/> is written to the datastore.
+		/// </summary>
+		public event EventHandler<BeforeWriteEventArgs> BeforeWrite;
```
Comment (antoineatrhea) [inline CDP4Dal/Session.cs]: Weird spacing in the file

File: CDP4JsonFileDal.NetCore.Tests/CDP4JsonFileDal.NetCore.Tests.csproj [hunk 9]
```diff
@@ -17,14 +17,15 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.0" />
+	  <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
```
Comment (antoineatrhea) [inline CDP4JsonFileDal.NetCore.Tests/CDP4JsonFileDal.NetCore.Tests.csproj:20]: Weird spacing

File: CDP4JsonFileDal.NetCore.Tests/JsonFileDalTestFixture.cs [hunk 10]
```diff
@@ -68,10 +66,12 @@ namespace CDP4JsonFileDal.NetCore.Tests
...
-        /// <summary>
-        /// AnnexC3 file archive
-        /// </summary>
-        private string annexC3File;
+	    private ILoggerFactory loggerFactory;
+
+		/// <summary>
+		/// AnnexC3 file archive
+		/// </summary>
+		private string annexC3File;
```
Comment (antoineatrhea) [inline CDP4JsonFileDal.NetCore.Tests/JsonFileDalTestFixture.cs]: Weird spacing in the file

File: CDP4JsonFileDal.Tests/CDP4JsonFileDal.Tests.csproj [hunk 11]
```diff
@@ -17,15 +17,16 @@
   </ItemGroup>
 
   <ItemGroup>
-	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.0" />
+	  <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
```
Comment (antoineatrhea) [inline CDP4JsonFileDal.Tests/CDP4JsonFileDal.Tests.csproj:20]: Weird spacing

File: CDP4JsonFileDal.Tests/JsonFileDalTestFixture.cs [hunk 12]
```diff
@@ -70,10 +68,12 @@ namespace CDP4JsonFileDal.Tests
     [TestFixture]
     public class JsonFileDalTestFixture
     {
-        /// <summary>
-        /// AnnexC3 file archive
-        /// </summary>
-        private string annexC3File;
+	    private ILoggerFactory loggerFactory;
+
```
Comment (antoineatrhea) [inline CDP4JsonFileDal.Tests/JsonFileDalTestFixture.cs:72]: Weird spacing in the file

File: CDP4JsonFileDal/CDP4JsonFileDal.csproj [hunk 13]
```diff
@@ -39,11 +39,11 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" />
-    <PackageReference Include="DotNetZip" Version="1.14.0" />
+	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" PrivateAssets="All" />
```
Comment (antoineatrhea) [inline CDP4JsonFileDal/CDP4JsonFileDal.csproj:42]: Weird spacing

File: CDP4JsonSerializer.Tests/CDP4JsonSerializer.Tests.csproj [hunk 14]
```diff
@@ -15,12 +15,12 @@
   </ItemGroup>
 
   <ItemGroup>
-	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" />
-	<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.0" />
-    <PackageReference Include="Moq" Version="4.18.3" />    
+	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" PrivateAssets="All" />
```
Comment (antoineatrhea) [inline CDP4JsonSerializer.Tests/CDP4JsonSerializer.Tests.csproj:18]: weird spacing

File: CDP4JsonSerializer.Tests/Deserializer/AliasDeserializerTestFixture.cs [hunk 15]
```diff
@@ -22,18 +21,19 @@
...
-using System.Globalization;
-using System.IO;
 
 namespace CDP4JsonSerializer.Tests.Deserializer
 {
     using System;
-    using System.Linq;
+    using System.Globalization;
+	using System.IO;
+	using System.Linq;
```
Comment (antoineatrhea) [inline CDP4JsonSerializer.Tests/Deserializer/AliasDeserializerTestFixture.cs]: weird spacing

File: CDP4JsonSerializer/Cdp4JsonSerializer.cs [hunk 16]
```diff
@@ -32,36 +32,42 @@ namespace CDP4JsonSerializer
...
-        /// The meta Info Provider.
-        /// </param>
-        /// <param name="supportedVersion">
-        /// The supported version of the data-model
-        /// </param>
-        public Cdp4JsonSerializer(IMetaDataProvider metaInfoProvider, Version supportedVersion)
+		/// <summary>
+		/// The <see cref="ILogger"/> used to log
+		/// </summary>
+		private readonly ILogger<Cdp4JsonSerializer> logger;
```
Comment (antoineatrhea) [inline CDP4JsonSerializer/Cdp4JsonSerializer.cs]: weird spacing in the file

File: CDP4JsonSerializer/JsonConverter/ClasslessDtoSerializer.cs [hunk 17]
```diff
@@ -22,52 +21,61 @@
...
+
     using CDP4JsonSerializer.Helper;
-    using Newtonsoft.Json;
-    using Newtonsoft.Json.Linq;
-    using NLog;
 
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Logging.Abstractions;
+
+	using Newtonsoft.Json;
```
Comment (antoineatrhea) [inline CDP4JsonSerializer/JsonConverter/ClasslessDtoSerializer.cs]: weird spacing in the file

File: CDP4JsonSerializer/JsonConverter/ThingSerializer.cs [hunk 18]
```diff
@@ -26,49 +26,54 @@ namespace CDP4JsonSerializer.JsonConverter
...
     /// </summary>
     public class ThingSerializer : JsonConverter
     {
-        /// <summary>
-        /// The NLog logger
-        /// </summary>
-        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+		/// <summary>
+		/// The <see cref="ILogger"/> used to log
+		/// </summary>
```
Comment (antoineatrhea) [inline CDP4JsonSerializer/JsonConverter/ThingSerializer.cs]: weird spacing in the file

File: CDP4ServicesDal.NetCore.Tests/CDP4ServicesDal.NetCore.Tests.csproj [hunk 19]
```diff
@@ -17,11 +17,12 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.0" />
-    <PackageReference Include="Moq" Version="4.18.3" />
+	<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
+	<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.5.0" />
```
Comment (antoineatrhea) [inline CDP4ServicesDal.NetCore.Tests/CDP4ServicesDal.NetCore.Tests.csproj:21]: weird spacing

File: CDP4ServicesDal.NetCore.Tests/CdpServicesDalTestFixture.cs [hunk 20]
```diff
@@ -72,10 +76,12 @@ public class CdpServicesDalTestFixture
         [SetUp]
         public void Setup()
         {
-            this.cancelationTokenSource = new CancellationTokenSource();
+	        this.loggerFactory = LoggerFactory.Create(builder => builder.AddConsole());
```
Comment (antoineatrhea) [inline CDP4ServicesDal.NetCore.Tests/CdpServicesDalTestFixture.cs]: weird spacing in the file

File: CDP4ServicesDal.NetCore.Tests/PostOperationTestFixture.cs [hunk 21]
```diff
@@ -27,19 +27,25 @@ namespace CDP4ServicesDal.Tests
...
     using System.Collections.Generic;
     using System.IO;
+
     using CDP4Common;
-    using CDP4Common.Dto;
-    using CDP4Common.MetaInfo;
-    using CDP4JsonSerializer;
+    using CDP4Common.CommonData;
+    using CDP4Common.DTO;
+	using CDP4Common.MetaInfo;
```
Comment (antoineatrhea) [inline CDP4ServicesDal.NetCore.Tests/PostOperationTestFixture.cs]: weird spacing in the file

File: CDP4ServicesDal.Tests/CDP4ServicesDal.Tests.csproj [hunk 22]
```diff
@@ -17,12 +17,13 @@
   </ItemGroup>
   
   <ItemGroup>
-	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" />
-	<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.0" />
-    <PackageReference Include="Moq" Version="4.18.3" />
+	<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
+	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" PrivateAssets="All" />
+	<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.5.0" />
```
Comment (antoineatrhea) [inline CDP4ServicesDal.Tests/CDP4ServicesDal.Tests.csproj:22]: weird spacing

File: CDP4ServicesDal.Tests/CdpServicesDalTestFixture.cs [hunk 23]
```diff
@@ -45,16 +45,20 @@ namespace CDP4ServicesDal.Tests
     using CDP4Dal.DAL.ECSS1025AnnexC;
     using CDP4Dal.Exceptions;
     using CDP4Dal.Operations;
-    
-    using NUnit.Framework;
+
+	using Microsoft.Extensions.Logging;
```
Comment (antoineatrhea) [inline CDP4ServicesDal.Tests/CdpServicesDalTestFixture.cs]: weird spacing in the file

File: CDP4ServicesDal.Tests/PostOperationTestFixture.cs [hunk 24]
```diff
@@ -27,19 +27,25 @@ namespace CDP4ServicesDal.Tests
     using System;
     using System.Collections.Generic;
     using System.IO;
+
     using CDP4Common;
-    using CDP4Common.Dto;
+	using CDP4Common.CommonData;
```
Comment (antoineatrhea) [inline CDP4ServicesDal.Tests/PostOperationTestFixture.cs]: weird spacing

File: CDP4ServicesDal/CDP4ServicesDal.csproj [hunk 25]
```diff
@@ -24,7 +24,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" />
+		<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" PrivateAssets="All" />
```
Comment (antoineatrhea) [inline CDP4ServicesDal/CDP4ServicesDal.csproj:27]: weird spacing

File: CDP4WspDal.NetCore.Tests/CDP4WspDal.NetCore.Tests.csproj [hunk 26]
```diff
@@ -17,11 +17,12 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.0" />
-    <PackageReference Include="Moq" Version="4.18.3" />
+	<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
+	<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.5.0" />
```
Comment (antoineatrhea) [inline CDP4WspDal.NetCore.Tests/CDP4WspDal.NetCore.Tests.csproj:21]: weird spacing

File: CDP4WspDal.NetCore.Tests/PostOperationTestFixture.cs [hunk 27]
```diff
@@ -27,17 +27,23 @@ namespace CDP4WspDal.Tests
...
-    using CDP4Common.Dto;
+	using CDP4Common.CommonData;
+	using CDP4Common.DTO;
     using CDP4Common.MetaInfo;
-    using CDP4JsonSerializer;
+	using CDP4Common.Types;
+	
     using CDP4Dal.Operations;
+
+	using CDP4JsonSerializer;
```
Comment (antoineatrhea) [inline CDP4WspDal.NetCore.Tests/PostOperationTestFixture.cs]: weird spacing

File: CDP4WspDal.NetCore.Tests/WSPDalTestFixture.cs [hunk 28]
```diff
@@ -57,7 +60,9 @@ namespace CDP4WspDal.Tests
     [TestFixture]
     public class WspDalTestFixture
     {
-        private WspDal dal;
+	    private ILoggerFactory loggerFactory;
```
Comment (antoineatrhea) [inline CDP4WspDal.NetCore.Tests/WSPDalTestFixture.cs]: weird spacing

File: CDP4WspDal.Tests/CDP4WspDal.Tests.csproj [hunk 29]
```diff
@@ -17,12 +17,13 @@
   </ItemGroup>
 
   <ItemGroup>
-	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" />
-	<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.0" />
-    <PackageReference Include="Moq" Version="4.18.3" />
+	<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
+	<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" PrivateAssets="All" />
+	<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.5.0" />
```
Comment (antoineatrhea) [inline CDP4WspDal.Tests/CDP4WspDal.Tests.csproj:22]: weird spacing

File: CDP4WspDal.Tests/PostOperationTestFixture.cs [hunk 30]
```diff
@@ -27,19 +27,25 @@ namespace CDP4WspDal.Tests
...
+
     using CDP4Common;
-    using CDP4Common.Dto;
+    using CDP4Common.CommonData;
     using CDP4Common.DTO;
     using CDP4Common.MetaInfo;
-    using CDP4JsonSerializer;
+    using CDP4Common.Types;
+
+	using CDP4JsonSerializer;
```
Comment (antoineatrhea) [inline CDP4WspDal.Tests/PostOperationTestFixture.cs]: weird spacing

File: CDP4WspDal.Tests/SessionTestFixture.cs [hunk 31]
```diff
@@ -1,6 +1,6 @@
 ﻿// --------------------------------------------------------------------------------------------------------------------
 // <copyright file="SessionTestFixture.cs" company="RHEA System S.A.">
-//    Copyright (c) 2015-2022 RHEA System S.A.
+//    Copyright (c) 2015-using CDP4Common.CommonData; RHEA System S.A.
```
Comment (antoineatrhea) [inline CDP4WspDal.Tests/SessionTestFixture.cs:3]: Replace using statement by the year

File: CDP4WspDal.Tests/WSPDalTestFixture.cs [hunk 32]
```diff
@@ -46,8 +46,10 @@ namespace CDP4WspDal.Tests
     using CDP4Dal.Operations;
     
     using CDP4WspDal;
-    
-    using NUnit.Framework;
+
+	using Microsoft.Extensions.Logging;
```
Comment (antoineatrhea) [inline CDP4WspDal.Tests/WSPDalTestFixture.cs]: weird spacing in the file

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
Spot other weird spacing and one error inside a file header

**Pull request merged**